src/host/PtySignalInputThread.cpp

```diff
@@ -147,6 +173,38 @@ void PtySignalInputThread::_DoClearBuffer()
     _pConApi->ClearBuffer();
 }
 
+// Method Description:
+// - Update the owner of the pseudo-window we're using for the conpty HWND. This
+//   allows to mark the pseudoconsole windows as "owner" by the terminal HWND
+//   that's actually hosting them.
+// - Refer to GH#2988
+// Arguments:
+// - data - Packet information containing owner HWND information
+// Return Value:
+// - <none>
+void PtySignalInputThread::_DoSetWindowParent(const SetParentData& data)
+{
+    _pConApi->ReparentWindow(data.handle);
+}
+
+// Method Description:
+// - If we were given a owner HWND, then manually start the pseudo window now.
+//   We need to do this specifically on the thread with the message pump. If the
+//   window is created on another thread, then the window won't have a message
+//   pump associated with it, and a DPI change in the connected terminal could
+//   end up HANGING THE CONPTY (for example).
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void PtySignalInputThread::StartPseudoWindowIfNeeded()
+{
+    if (_earlyReparent.has_value())
+    {
+        ServiceLocator::LocatePseudoWindow(reinterpret_cast<HWND>(_earlyReparent.value().handle));
+    }
+}

```

Comment:
This function is redundant with `ConnectConsole()`.
You can just move this if condition in there, because `ConnectConsole()` is only called in one place, which coincidentally is also the only place `StartPseudoWindowIfNeeded()` is called.

Can't forget to call both functions if there's just one. \*taps forehead\*

Reply:
huh. what, weird. Maybe I wasn't doing this bit in `ConnectConsole` in an earlier commit, which made this initially necessary.